modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/genesis/GenesisFSSuite.scala

```diff
@@ -0,0 +1,70 @@
+package org.tessellation.sdk.infrastructure.genesis
+
+import cats.effect.{IO, Resource}
+import eu.timepit.refined.auto._
+import fs2.io.file.Files
+import fs2.text
+import org.scalacheck.Gen
+import org.tessellation.currency.schema.currency.CurrencySnapshot
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.schema.generators.{addressGen, balanceGen}
+import org.tessellation.sdk.sdkKryoRegistrar
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.{KeyPairGenerator, SecurityProvider}
+import weaver._
+import weaver.scalacheck._
+
+object GenesisFSSuite extends MutableIOSuite with Checkers {
+  type Res = (KryoSerializer[IO], SecurityProvider[IO])
+
+  def sharedResource: Resource[IO, Res] = for {
+    ks <- KryoSerializer.forAsync[IO](sdkKryoRegistrar)
+    sp <- SecurityProvider.forAsync[IO]
+  } yield (ks, sp)
+
+  val gen: Gen[(Map[Address, Balance], Address)] = for {
+    balancesMap <- Gen.mapOf(balanceGen.flatMap(balance => addressGen.map((_, balance))))
+    identifier <- addressGen
+  } yield (balancesMap, identifier)
+
+  test("writes and loads genesis") { res =>
+    implicit val (ks, sp) = res
+    val loader = GenesisFS.make[IO, CurrencySnapshot]
+
+    forall(gen) {
+      case (balances, identifier) =>
+        for {
+          tempDir <- Files[F].createTempFile
```

Comment:
`createTempFile` won't clean up after its done.

```suggestion
        Files[F].tempFile.use { tmpPath =>
          for {
```
and add closing brace. I also renamed the variable, if you agree it's clearer.

Reply:
Good point. Using Resource is much better